uniffi_bindgen/src/interface/conventions.rs

```diff
@@ -0,0 +1,144 @@
+use crate::interface::{Enum, FfiDefinition, Record};
+use crate::{ComponentInterface, VisitMut};
+use std::collections::{BTreeMap, BTreeSet};
+use uniffi_meta::Type;
+
+impl ComponentInterface {
+    /// Walk down the [`ComponentInterface`] and adjust the names of each type
+    /// based on the naming conventions of the supported languages.
+    ///
+    /// Each suppoerted language implements the [`VisitMut`] Trait and is able
+    /// to alter the functions, enums etc. to its own naming conventions.
+    pub fn visit_mut<V: VisitMut>(&mut self, visitor: &V) {
+        for type_ in self.types.type_definitions.values_mut() {
+            visitor.visit_type(type_);
+        }
+
+        let mut all_known_types_altered: BTreeSet<Type> = BTreeSet::new();
+
+        for type_ in self.types.all_known_types.iter() {
```

Comment:
The only way to alter a `BTreeSet` is to take out the element (via `self.types.all_known_types.get()`), mutate it, and insert it back. Since we are modifying all elements, it is more efficient to just create a new `BTreeSet` all together. AFAIK, there is no `iter_mut` implementation for this structure. But maybe I am missing something!

Reply:
Nope, I was missing that this was a BTreeSet and not a BTreeMap (even though I typed out the type name :man_facepalming: )